src/main.ts

```diff
@@ -153,6 +153,12 @@ async function runWorkflow(
   const steps: Steps = {};
   steps.build = await build(bufPath, inputs);
   if (steps.build.status == Status.Failed) {
+    if (steps.build.stderr.match(/had no .proto files/)) {
+      core.info(
+        "Empty repository detected, ensure the repository is checked out",
```

Comment:
If we are just matching against this error string, we haven't actually verified anything more than there are no proto files (aka we haven't verified you forgot to checkout a repository in the first place).

I think there are two paths we could do down.
1. Just check the error string but then suggest a possible cause (e.g. "Did you forget to add the `actions/checkout@v4` step to your workflow?")
2. Actually check to see if if the pwd is a git repo. This would then allow us to differentiate between "you forgot to checkout the repo" from "you checked out the repo and there were no files in it"

Reply:
The input can be any input, so defer to the error to avoid parsing the input, ie. could be a remote `https://github.com/googleapis/googleapis.git`. We also have a usecase of just wanting to install `buf` with `setup-only`, so would want to only check the repo if building on the condition the input is empty (to avoid parsing it).

1. Will do this for now.
2. Don't check for local repo to avoid use cases where there is intentionally no local repository.

We could also make the input a .git reference like the breaking_against input, avoiding the checkout step. As an alternative solution.

Reply:
This suggestion really only makes sense for local input paths (not remote ones) so we should check for that and only show this if there is not a remote input path

Reply:
Implemented no.1 for now, fixed in [69a7b9d](https://github.com/bufbuild/buf-action/pull/62/commits/69a7b9d1286123dd4833827d5f68ff76e01b1867)